Client/Frontend/Browser/Tabs/RemoteTabs/RemoteTabsEmptyView.swift

```diff
@@ -0,0 +1,137 @@
+    struct UX {
+        static let verticalPadding: CGFloat = 40
+        static let horizontalPadding: CGFloat = 24
+        static let paddingInBetweenItems: CGFloat = 15
+        static let buttonCornerRadius: CGFloat = 13
+        static let titleSizeFont: CGFloat = 22
+        static let descriptionSizeFont: CGFloat = 17
+        static let buttonSizeFont: CGFloat = 16
+        static let imageSize = CGSize(width: 90, height: 60)
+        static let buttonVerticalInset: CGFloat = 12
+    }
+
+    weak var delegate: RemotePanelDelegate?
+
+    // MARK: - UI
+
+    private lazy var stackView: UIStackView = .build { stackView in
+        stackView.axis = .vertical
+        stackView.alignment = .fill
+        stackView.distribution = .fillProportionally
+        stackView.spacing = UX.paddingInBetweenItems
+        stackView.alignment = .center
+    }
+
+    private let emptyStateImageView: UIImageView = .build { imageView in
+        imageView.contentMode = .scaleAspectFit
+    }
+
+    private let titleLabel: UILabel = .build { label in
+        label.adjustsFontForContentSizeCategory = true
+        label.font = DefaultDynamicFontHelper.preferredFont(withTextStyle: .title2,
+                                                            size: UX.titleSizeFont)
+        label.numberOfLines = 0
+        label.textAlignment = .center
+    }
+
+    private let instructionsLabel: UILabel = .build { label in
+        label.adjustsFontForContentSizeCategory = true
+        label.font = DefaultDynamicFontHelper.preferredFont(withTextStyle: .body,
+                                                            size: UX.descriptionSizeFont)
+        label.numberOfLines = 0
+        label.textAlignment = .center
+    }
+
+    private let signInButton: ResizableButton = .build { button in
+        button.titleLabel?.font = DefaultDynamicFontHelper.preferredFont(withTextStyle: .callout,
+                                                                         size: UX.buttonSizeFont)
+        button.setTitle(.Settings.Sync.ButtonTitle, for: [])
+        button.layer.cornerRadius = UX.buttonCornerRadius
+        button.contentEdgeInsets = UIEdgeInsets(top: UX.buttonVerticalInset,
+                                                left: UX.buttonVerticalInset,
+                                                bottom: UX.buttonVerticalInset,
+                                                right: UX.buttonVerticalInset)
+        button.accessibilityIdentifier = AccessibilityIdentifiers.TabTray.syncDataButton
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setupLayout()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func configure(state: RemoteTabsPanelEmptyState,
+                   theme: Theme?,
+                   delegate: RemotePanelDelegate?) {
+        self.delegate = delegate
+
+        if let theme = theme {
```

Comment:
@OrlaM @mattreaganmozilla We didn't reach a decision in our last conversation about handling theming with Redux but I feel we were leaning against it at least on the first pass of the refactoring

Reply:
@yoanarios Ok thanks; I wasn't sure what the plan was, so just injected/applied current theme during configuration for now. Happy to revisit this as needed, or whatever is we think will be best; I thought perhaps theming would come later once we can test/exercise the new UI components a bit more thoroughly (so it's easier to catch any scenarios where the theming isn't applied correctly etc.)

Reply:
> I think we should continue using Theme as the rest of the app for now so we can remove the TODO and the optional Theme in the configure function.

Can you clarify this a bit further? Or is there a specific file I can reference for the pattern you're wanting? Thanks @yoanarios

Reply:
You can take any view conforming to `ThemeApplicable` like `ActionFooterView`.

The view needs to conform to `ThemeApplicable` and add `func applyTheme(theme: Theme)` then we need to call applyTheme where the view is initialized passing the currentTheme

Reply:
> The view needs to conform to ThemeApplicable and add func applyTheme(theme: Theme)

It already does, so I'm not clear if there's anything else needed for this? I had previously added a ToDo to update to Redux-based theming at some point but until then I have `applyTheme` implemented here as it is for most of the other `ThemeApplicable` views.

Reply:
Thanks @yoanarios , should be addressed now by https://github.com/mozilla-mobile/firefox-ios/pull/16688/commits/76a01cdb405bbd290c2662a207558cd630b3eba5